cartographer/mapping/trajectory_builder_interface.h

```diff
@@ -44,14 +44,20 @@ proto::TrajectoryBuilderOptions CreateTrajectoryBuilderOptions(
 // optimized pose estimates.
 class TrajectoryBuilderInterface {
  public:
+  struct InsertionResult {
+    std::shared_ptr<const mapping::TrajectoryNode::Data> constant_data;
+    std::vector<std::shared_ptr<const Submap>> insertion_submaps;
+  };
```

Comment:
Wouldn't it make sense to include the `NodeId` here as well and just pass one struct?

Reply:
Is it true that you have a `NodeId` iff you have an `InsertionResult`.

Reply:
I think so. The comment below also says that.